CMakeLists.txt

```diff
@@ -0,0 +1,187 @@
+
+if(GIT_FOUND)
+  execute_process(
+    COMMAND ${GIT_EXECUTABLE} log -1 --pretty=format:%ct
+    WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
+    OUTPUT_VARIABLE COMPTIME
+    RESULT_VARIABLE GIT_RESULT
+    ERROR_QUIET
+    OUTPUT_STRIP_TRAILING_WHITESPACE
+  )
+  if(NOT GIT_RESULT EQUAL 0)
+    set(COMPTIME "0000000000")
+  endif()
+
+  execute_process(
+    COMMAND ${GIT_EXECUTABLE} log -1 --pretty=format:%D
+    WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
+    OUTPUT_VARIABLE VERSION_EXPORT
+    RESULT_VARIABLE GIT_RESULT
+    ERROR_QUIET
+    OUTPUT_STRIP_TRAILING_WHITESPACE
+  )
+  if(NOT GIT_RESULT EQUAL 0)
+    set(VERSION_EXPORT "HEAD -> master")
+  endif()
+else(GIT_FOUND)
+  set(COMPTIME "0000000000")
+  set(VERSION_EXPORT "HEAD -> master")
+endif(GIT_FOUND)
+set(GoodbyeDPI_VERSION "${VERSION_EXPORT} ${VERSION_FROM_GIT}")
+message(STATUS "version tag: ${GoodbyeDPI_VERSION}")
+
+if(${CMAKE_SYSTEM_NAME} MATCHES "Darvin")
+    set(MACOS TRUE)
+elseif(${CMAKE_SYSTEM_NAME} MATCHES "Linux")
+    set(LINUX TRUE)
+elseif(${CMAKE_SYSTEM_NAME} MATCHES "FreeBSD")
+    set(FREEBSD TRUE)
+elseif(${CMAKE_SYSTEM_NAME} MATCHES "Windows")
+    set(WINDOWS TRUE)
+endif()
+
+set(Source_dir "${CMAKE_CURRENT_SOURCE_DIR}/src")
+set(Utils_dir "${CMAKE_CURRENT_SOURCE_DIR}/src/utils")
+set(WinDivert_dir "${CMAKE_CURRENT_SOURCE_DIR}/WinDivert")
+set(CMake_Misc_Dir "${CMAKE_CURRENT_SOURCE_DIR}/cmake")
+
+file(GLOB_RECURSE SRCFILES "${Source_dir}/*.c" "${Source_dir}/*.cpp")
+
+include_directories(GoodbyeDPI "${WinDivert_dir}/include")
+if(MSVC)
+    set(divertCompilerName "WDDK")
+elseif(MINGW)
+    set(divertCompilerName "MINGW")
+endif()
+set(divertLibraryPath "${WinDivert_dir}/install/${divertCompilerName}/${CMAKE_SYSTEM_PROCESSOR}/")
+message(STATUS ${divertLibraryPath})
+
+#making resource file
+set(rc_script "${Source_dir}/goodbyedpi-rc.rc")
+set(rc_script_proto "${CMake_Misc_Dir}/goodbyedpi-rc.rc.in")
+set(license_file "${CMAKE_CURRENT_SOURCE_DIR}/LICENSE")
+file(READ ${license_file} LICENSE_TEXT)
+string(REPLACE "\"" "\"\"" LICENSE_TEXT ${LICENSE_TEXT})
+string(REPLACE "\r\n" "\n" LICENSE_TEXT ${LICENSE_TEXT})
+string(REPLACE "\n" "\\r\\n" LICENSE_TEXT ${LICENSE_TEXT})
+message(STATUS "generating resources: ${rc_script_proto} -> ${rc_script}")
+configure_file(${rc_script_proto} ${rc_script})
+file(GLOB_RECURSE resource_files "${Source_dir}/*.rc" "${Source_dir}/*.ico")
+
+
+if(MSVC_IDE)
+   file(GLOB_RECURSE HDRFILES "${Source_dir}/*.h")
+endif()
+
+# config.h
+set(CONFIG_FILENAME "config.h")
+set(CONFIG_FILE "${Source_dir}/${CONFIG_FILENAME}")
+mark_as_advanced(CONFIG_FILE)
+set(CONFIG_FILE_PROTO "${CMake_Misc_Dir}/${CONFIG_FILENAME}.in")
+
+set(GoodbyeDPI_SERVICE_NAME "${PACKAGE_NAME}" CACHE STRING "Service name")
+
+set(GoodbyeDPI_HOST_MAXLEN 253 CACHE INTEGER "Max length of a host")
+set(GoodbyeDPI_MAX_FILTERS 4 CACHE INTEGER "Max length of a host")
+set(GoodbyeDPI_MAX_PACKET_SIZE 9016 CACHE INTEGER "Max length of a host")
```

Comment:
It's always a good idea to have everything hardcoded configurable.

Reply:
Maximum host size is defined in RFC. Max filters should not be changed since it's internal variable for me. Packet size should not be configured in a normal circumstances either.